arch/xtensa/src/common/xtensa_hostfs.c

```diff
@@ -114,6 +119,18 @@ ssize_t host_read(int fd, void *buf, size_t count)
   up_invalidate_dcache(buf, buf + count);
 #endif
 
+  if (fd == STDIN_FILENO)
```

Comment:
do you mean STDIN_FILENO can be returned by SIMCALL_SYS_OPEN?
how can it happen?

Reply:
No, SIMCALL_SYS_OPEN never returns id in range 0-2, so host_read can reuse STDIN_FILENO to indicate the special handle.

Reply:
> SIMCALL_SYS_OPEN never returns id in range 0-2

are you assuming a particular implementation of SIMCALL_SYS_OPEN?

Reply:
Yes, we try the implementation on Cadance xtensa emulator and Linux simulator.

Reply:
i don't understand why you need to open the device this way.
it's cleaner to make uart_hostfs call host_open explicitly to open the device.